File: java/P28.java
class Solution {
    public int strStr(String haystack, String needle) {
        if (haystack == null) return -1;
        if (needle == null) return 0;
        int l1 = haystack.length();
        int l2 = needle.length();
        if (l1 < l2) return -1;
        int diff = l1-l2;
        for (int i = 0; i <= diff; i++) {
            if (haystack.substring(i, i+l2).equals(needle)) {
                return i;
            }
        }
        return -1;
    }
}

File: python/P28.py
class Solution(object):
    def strStr(self, haystack, needle):
        if not haystack and not needle:
            return 0
        if len(needle) > len(haystack):
            return -1
        len_h = len(haystack)
        len_n = len(needle)
        for i in range(len_h-len_n+1):
            if haystack[i:i+len_n] == needle:
                return i
        return -1
        